Alternative Pepper Spray rework, more balanced.

MacBlaze1 opened the pull request:
closes #46269

This PR is designed to make pepper spray a useful and viable tool of the security arsenal. Oranges has said he likes sec buffs, and this will help sec use pepper spray instead of just throwing it away to store another e-bola. I will close this PR if carlarc brings his PR more in line. 



## About The Pull Request
Volume increased to 50 for pepper spray, added knockdown of decreasing degree for decreasing importance pepper spray entrance organs, added confuse, blind, blur, slowdown.

## Why It's Good For The Game

Pepper spray is not even worth using right now, and I routinely see sec officers throwing away their pepper spray for e-bolas, handcuffs or flashbangs. This PR will make pepper spray a viable option in the sec arsenal, but does account for the range and spam ability of pepper spray while still making it competitive. As we are moving away from hardstuns, I see it fit to replace the hard stun with a soft stamina loss and knockdown. Gas masks and only gas masks block pepper spray, this is an exact parallel to flashes and sunglasses. 
## Changelog
:cl:
balance: increased volume of pepper sprays to 50 
tweak: adds knockdown of 3 second for no protection, 3 for no mouth protection
tweak: Changes bluring and blindness of eyes
balance: Replaces Paralysis with blind
balance: gas masks now are the pepper spray equivalent of sunglasses, only much much easier to get.
/:cl:

MacBlaze1 committed:
Initial changes to pepper spray

Volume increased to 50 for pepper spray, added knockdown of decreasing degree for decreasing importance pepper spray entrance organs, adjusted stamina damage to account for spray range and quick spraying, Added confused effects, adjusted overlay damage to numbers more inline with the stamina loss

MacBlaze1 committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into Pepperspray-rework

MacBlaze1 committed:
Small fix

removes mouth confusion

MacBlaze1 committed:
more small tweaks

adds a comment

wesoda25 commented:
Why does it do any stamina damage at all? Thats stupid and doesn’t make sense. Pepperspray should be a deterrent, not a reliable stun weapon. A short knockdown + blind scaling with how much of the spray is blocked by clothing would work better. Sec can always follow up with a REAL weapon if they want.

MacBlaze1 commented:
> Why should it do any stamina damage at all? Thats stupid and doesn’t make sense. Pepperspray should be a deterrent, not a reliable stun weapon. A short knockdown + blind scaling with how much of the spray is blocked by clothing would work better. Sec can always follow up with a REAL weapon if they want.

A lot of shit doesnt make sense in ss13. Pepper spray IS a reliable stun weapon. You can hard stun lock someone with the pepper spray we have now. Let me reduce stamina damage.

MacBlaze1 committed:
Reduction of Stamina damage

Changes Stamina damage from 40 to 25 for completely unprotected, 25 to 15 for no eye  protection and 15 to 10 for no mouth protection.

wesoda25 commented:
I know that. I’m saying we should just remove the stun, give a knockdown (maybe increase blind) and be done with it

MacBlaze1 commented:
> I know that. I’m saying we should just remove the stun, give a knockdown (maybe increase blind) and be done with it

The thing is, this is all countered by any eyewear and anything covering your mouth. Ill think about increasing blindness. What would you recommend for the blind duration?

Shadowflame909 commented:
@MacBlaze1 No Knock-Down but permanent blindness unless you wash your face/get in the shower

Suffer with the eye stinging pain

wesoda25 commented:
Nah. Make if eyes are covered no blind, but a 3 secondish knockdown. If mouth is covered a 5-10 second blind plus welding visor vision until you wash your eyes (good idea shadow). If nothing is covered just combine the two.

carlarctg commented:
idk how to code eye clean so if yall do that i'll close my pr

GuyonBroadway commented:
Now do what hippiestation did and let officers fill up fire extinguishers from pepper spray dispensers.

MacBlaze1 commented:
> @MacBlaze1 No Knock-Down or stamina but 3-minute blindness unless you wash your face/get in the shower
> 
> Also hit them with a major bad moodlet like "AHH I CANT SEE" or "MY EYES!!"
> 
> Suffer with the eye stinging pain

What value should the moodlet be?

MacBlaze1 committed:
Accepted Suggestions

Increases eye blindness to 3 minutes unless you wash yourself and knockdown to 5 seconds without any protection and 3 seconds with no mouth protection.

MacBlaze1 committed:
Added slowdown

30 second slowdown if you dont have mouth protection

MacBlaze1 committed:
Trying to fix errors

shit

MacBlaze1 committed:
Attempted fix 2

Fixes the damn types and errors

MacBlaze1 committed:
I think I found it

CPTANT commented:
I say: blind + stumble + drop items

makes it great for self defence, but hard to put someone down with.

MacBlaze1 committed:
Hopefully fixes

MacBlaze1 committed:
Hopeful 2

MacBlaze1 committed:
more fixes

MacBlaze1 committed:
more changes rebalances

MacBlaze1 committed:
more balance changes

MacBlaze1 committed:
Update food_reagents.dm

nemvar commented on code/modules/reagents/reagent_containers/spray.dm:
```diff
@@ -197,7 +197,7 @@
+	volume = 50
 	stream_range = 4
 	amount_per_transfer_from_this = 5
 	list_reagents = list(/datum/reagent/consumable/condensedcapsaicin = 40)
```
This also needs changing if you want to increase the volume unless you want to force people to refill their initial pepperspray for whatever reason

MacBlaze1 replied:
ah thanks nemvar!

MacBlaze1 committed:
This is where the fun begins

Ill try balancing around increasing time means decreasing severity of effects.

MacBlaze1 committed:
small fix

MacBlaze1 committed:
moar changes

MacBlaze1 committed:
Please work

MacBlaze1 committed:
finally fixed

MacBlaze1 committed:
updated starting volume

Shadowflame909 commented:
@MacBlaze1 the same type of negative moodlet you get when your arm gets chopped off. But this one goes away faster IE is more temporary. Since you can wash your eyes.

MacBlaze1 commented:
> @MacBlaze1 the same type of negative moodlet you get when your arm gets chopped off. But this one goes away faster IE is more temporary. Since you can wash your eyes.

I have no idea what the fuck code is needed or what signal is sent when you are wet or wash eyes.

MacBlaze1 committed:
Update food_reagents.dm

MacBlaze1 committed:
rebalances gas masks

wesoda25 commented:
VERIFIED

MacBlaze1 committed:
increases to slowdown

from 10% to 25%

MacBlaze1 committed:
Update .gitignore

MacBlaze1 committed:
Update game_options.txt

MacBlaze1 committed:
Update .gitignore

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/food_reagents.dm:
```diff
@@ -283,33 +283,33 @@
+		
 		//actually handle the pepperspray effects
-		if ( eyes_covered && mouth_covered )
+		if (eyes_covered && istype(victim.wear_mask, /obj/item/clothing/mask/gas))
```
```DM

if(istype(victim.wear_mask, /obj/item/clothing/mask/gas))
    if(eyes_covered)
        return
    else
        if(prob(5))
            scream
        rest of code
```

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/medicine_reagents.dm:
```diff
@@ -1234,3 +1234,12 @@
+// BEGIN_PREFERENCES
+// END_PREFERENCES
+// BEGIN_INCLUDE
+// END_INCLUDE
```
wut

MacBlaze1 replied:
I have no idea why the hell thats there, let me delete that

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/reagents/food_reagents.dm:
```diff
@@ -283,33 +283,33 @@
+		if (eyes_covered && istype(victim.wear_mask, /obj/item/clothing/mask/gas))
 			return
-		else if ( mouth_covered )	// Reduced effects if partially protected
+		else if(istype(victim.wear_mask, /obj/item/clothing/mask/gas))	// Reduced effects if partially protected
```
prob should readd mouth_covered and just change it to this proc instead of `is_mouth_covered`

MacBlaze1 replied:
Its supposed to be the sunglasses equivalent of pepperspray. Gas masks are, that is.

ExcessiveUseOfCobblestone commented:
are gasmasks considered eye protectors for the purpose of the check here?

MacBlaze1 committed:
Update medicine_reagents.dm

MacBlaze1 commented:
@ExcessiveUseOfCobblestone no, they protect only the mouth, but anything can protect the eyes.

nero1024 commented:
The gas masks cover the face, you would expect them to protect your eyes too.

carlarctg commented:
not enough tiders using gas masks lately, making them pepperproof will bring them back into the assistant meta

MacBlaze1 commented:
@carlarctg I dont think you understand @nero1024 or you They NEVER protected the eyes.

carlarctg commented:
make them eyeproof then :cat2:

ExcessiveUseOfCobblestone commented:
if they're not eye protection then the analogy of flashes:glasses::spray:gasmasks doesn't really hold true since gasmasks are half the battle.

MacBlaze1 commented:
> if they're not eye protection then the analogy of flashes:glasses::spray:gasmasks doesn't really hold true since gasmasks are half the battle.

@ExcessiveUseOfCobblestone Yes well Id like to keep the necessary eye protection, which could literally be science goggles or a medical HUD for god sakes. Eyewear is incredibly easy to get, and it protects against the blindness and confusion.

MacBlaze1 commented:
Dammit the movespeed_modifications is broke.

MacBlaze1 committed:
WORK DAMMIT

AHHHHHHHHHHHHHhh FIXES THE BULKY DRAGGING WHY THIS 1 LINE CHANGE UNGA BREAK PR

MacBlaze1 committed:
WHY

MacBlaze1 committed:
Merge branch 'master' into Pepperspray-rework

ExcessiveUseOfCobblestone commented on code/__DEFINES/movespeed_modification.dm:
```diff
@@ -1,69 +1,71 @@
+#define MOVESPEED_ID_BULKY_DRAGGING                     "BULKY_DRAG"
+#define MOVESPEED_ID_HUMAN_CARRYING                     "HUMAN_CARRY"
+#define MOVESPEED_ID_SHRINK_RAY							"SHRUNKEN_SPEED_MODIFIER"
+#define MOVESPEED_ID_PEPPER_SPRAY                       "PEPPER_SPRAYED"
```
wtf

MacBlaze1 replied:
I didnt want to do it reeee

MacBlaze1 replied:
All I wanted was a quick fix dammit

PKPenguin321 replied:
Do `git checkout upstream/master -- code/__DEFINES/movespeed_modification.dm` to undo changes to this file

MacBlaze1 replied:
Yeah Ill do git pull upstream master first then try that

ExcessiveUseOfCobblestone commented:
I'll try using the webbitor, that's a good trick!

MacBlaze1 committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into Pepperspray-rework

MacBlaze1 committed:
Update movespeed_modification.dm

MacBlaze1 committed:
Update movespeed_modification.dm

GuyonBroadway committed:
Makes felinids die from chocolate. (#46238)

* Makes felinids die from chocolate.

* tabs wew

* Return parent

* Moves code to felind.dm

* removes ded code from felinid.dm

tgstation-server committed:
Automatic changelog generation for PR #46238 [ci skip]

GuyonBroadway committed:
Makes it so chocolate doesn't last inside felinids forever. (#46339)

* Makes it so chocolate doesn;t last inside felinids forever.

* Guyon has high int but low wis

tralezab committed:
fixes infinite butter (#46302)

* relic file

* song for this fix: Jimmy Buffett - Margaritaville

* wow

* need to test first?

* Update snacks_other.dm

MacBlaze1 committed:
ahhh fuck

MacBlaze1 committed:
wew

MacBlaze1 committed:
wew

MacBlaze1 committed:
Revert "wew"

This reverts commit 769a376b661029d685671aeab1166ecf0cfd189e.

MacBlaze1 committed:
Revert "Revert "wew""

This reverts commit 9e071d92fdf157cccb48a7d3ddfd33f6224f1354.

MacBlaze1 committed:
Revert "Revert "Revert "wew"""

This reverts commit 9c37ef1d473adc5474ccc43bbd4d9628df3a090e.

MacBlaze1 committed:
Revert "Revert "Revert "Revert "wew""""

This reverts commit 4e5fba86d0f20438362a4a84936ed0d974bd3348.

MacBlaze1 committed:
Revert "wew"

This reverts commit 769a376b661029d685671aeab1166ecf0cfd189e.

MacBlaze1 committed:
Revert "Revert "wew""

This reverts commit d85a4d19b220655f25f7f80bddb85a70e6dcac59.

MacBlaze1 committed:
wew

MacBlaze1 committed:
wew

MacBlaze1 committed:
Merge branch 'Pepperspray-rework' of https://github.com/MacBlaze1/tgstation into Pepperspray-rework

MacBlaze1 commented:
Fucked up my repo, opening new one
